### PR TITLE
Add `Style/RedundantRegexpCharacterClass` cop

### PR DESCRIPTION
**Pull request by owst**
This PR adds a cop to check for unnecessary single-element Regexp character classes:

```ruby
  # This cop checks for unnecessary single-element Regexp character classes.
  #
  # @example
  #
  #   # bad
  #   r = /[x]/
  #
  #   # good
  #   r = /x/
  #
  #   # bad
  #   r = /[\s]/
  #
  #   # good
  #   r = /\s/
  #
  #   # good
  #   r = /[ab]/
```

-----------------

Before submitting the PR make sure the following are checked:

* [x] Wrote [good commit messages][1].
* [x] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [x] Squashed related commits together.
* [x] Added tests.
* [x] Added an entry to the [Changelog](https://github.com/rubocop-hq/rubocop/blob/master/CHANGELOG.md) if the new code introduces user-observable changes. See [changelog entry format](https://github.com/rubocop-hq/rubocop/blob/master/CONTRIBUTING.md#changelog-entry-format).
* [x] The PR relates to *only* one subject with a clear title and description in grammatically correct, complete sentences.
* [x] Run `bundle exec rake default`. It executes all tests and RuboCop for itself, and generates the documentation.

[1]: https://chris.beams.io/posts/git-commit/

### CODE REVIEW

**Comment by koic** [comment]
It sounds like a good idea, but maybe it's better to suggest it to Rubocop::RegularExpression.
https://github.com/pocke/rubocop-regular_expression

cc @pocke

**Comment by owst** [comment]
Hmm - is `rubocop-regular_expression` released? It looks a little like a work in progress (but I like the cop it adds 👍 ). 

I'm also proposing to add another related Regexp cop in #7908 and Rubocop already has 
```
Lint/AmbiguousRegexpLiteral
Lint/RegexpAsCondition
Lint/UriRegexp
Style/RegexpLiteral
```
which are Regexp-related.

What do you think about adding this (and #7908) and maybe(?) later extracting them into `rubocop-regular_expression` if it is released?

**Comment by bbatsov** [comment]
For the record - I'm fine with adding more regular expression cops to the main gem. Looking at https://github.com/rubocop-hq/rubocop/pull/7749 I assume that @pocke is fine with this approach as well.

**Comment by marcandre** [comment]
I also wouldn't have regexp cops separated.

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,7 @@
 ## master (unreleased)
 
 ### New features
+* [#8055](https://github.com/rubocop-hq/rubocop/pull/8055): Add new `Style/RedundantRegexpCharacterClass` cop. ([@owst][])
```
Comment (bbatsov) [inline CHANGELOG.md:6]: Can you add a blank line after the heading and remove the blank line between this and the next changelog entry?

File: config/default.yml [hunk 2]
```diff
@@ -3607,6 +3607,11 @@ Style/RedundantPercentQ:
   Enabled: true
   VersionAdded: '0.76'
 
+Style/RedundantRegexpCharacterClass:
+  Description: 'Checks for unnecessary single-element Regexp character classes.'
+  Enabled: true
```
Comment (bbatsov) [inline config/default.yml:3612]: true -> pending

File: lib/rubocop/cop/style/redundant_regexp_character_class.rb [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+      #   # good
+      #   r = /\s/
+      #
+      #   # good
+      #   r = /[ab]/
+      class RedundantRegexpCharacterClass < Cop
+        include MatchRange
+
+        MSG_REDUNDANT_CHARACTER_CLASS =
+          'Redundant single-element character class, `[` and `]` can be removed'
```
Comment (bbatsov) [inline lib/rubocop/cop/style/redundant_regexp_character_class.rb:28]: You forgot the final `.`. I'd also make this more actionable and word it like `Redundant ... [x] can be replaced with x.` (or something along those lines).

**Comment by bbatsov** [comment]
Great cop, btw! I love it!

**Comment by koic** [comment]
> Looking at #7749 I assume that @pocke is fine with this approach as well.

I forgot #7749. Certainly this cop could be included in the core. Thank you!

**Comment by bbatsov** [comment]
@owst You also have to rebase on top of `master`.

**Comment by bbatsov** [comment]
@owst The CI is failing now.

**Comment by owst** [comment]
@bbatsov I needed a tweak to the cop added in #7908 - see #8073

**Comment by bbatsov** [comment]
@owst I guess you have to rebase and squash here and we're good to go.

**Comment by owst** [comment]
Done @bbatsov

**Comment by bbatsov** [comment]
Thanks!

**Pull request merged**